docs/source/installation/development.rst

```diff
@@ -3,5 +3,123 @@
 Installation guide (development)
 ===============================
 
-.. todo::
-    Write guide
+Install System Packages
+-----------------------
+
+We'll assume you've installed PostgreSQL (and its ``contrib package``, if on a Linux system) and managed to setup/start
+the server. PostgreSQL 9.6 at least is required. You can find instructions on how to install it on Fedora/CentOS
+`here <https://www.if-not-true-then-false.com/2012/install-postgresql-on-fedora-centos-red-hat-rhel/>`_.
+You can find the same for Debian/Ubuntu `here <https://www.tecmint.com/install-postgresql-on-debian-and-ubuntu/>`_.
+
+CentOS/Fedora
++++++++++++++
+
+.. code-block:: shell
+
+    yum install -y gcc redis python-devel python-virtualenv libjpeg-turbo-devel libxslt-devel libxml2-devel \
+        libffi-devel pcre-devel libyaml-devel redhat-rpm-config
+    systemctl start redis.service
+
+
+Debian/Ubuntu
++++++++++++++
+
+.. code-block:: shell
+
+    apt install -y --install-recommends python-dev python-virtualenv libxslt1-dev libxml2-dev libffi-dev libpcre3-dev \
+        libyaml-dev build-essential redis-server
+
+Then on Debian::
+
+    apt install -y libjpeg62-turbo-dev
+
+And on Ubuntu::
+
+    apt install -y libjpeg-turbo8-dev zlib1g-dev
+
+
+Creating the directory structure
+--------------------------------
+
+You will need a directory in your file system to store Indico as well as its data files (archives, etc...). Some
+developers keep all their code inside a ``dev`` or ``code`` dir. We will assume ``dev`` here.
+
+.. code-block:: shell
+
+    mkdir -p ~/dev/indico/data
+
+We will need a virtualenv where to run Indico::
+
+    cd ~/dev/indico
+    virtualenv env -p /usr/bin/python2.7
+
+
+Cloning Indico
+--------------
+
+First, let's clone Indico's code base. If you're going to contribute back to the project, it's probably best if
+you clone your own `GitHub fork of the project <https://help.github.com/articles/fork-a-repo/>`_. Otherwise, if you're
+just thinking of playing a bit with the source code, it's enough if you clone the upstream repo:
+
+.. code-block:: shell
+
+    git clone --recursive https://github.com/indico/indico.git src
+
+If you're going to be changing the standard Indico plugins and/or the documentation, you can also clone those::
+
+    git clone https://github.com/indico/indico-plugins.git plugins
```

Comment:
I'd add a subfolder `plugins/something` for people who'll also work on their own plugins (or clone the CERN plugins in case of team members). Like this you can open "plugins" as another project in your IDE and have all plugins there, no matter from which repo.